contrib_bots/lib/commute_bot.py

```diff
@@ -0,0 +1,262 @@
+        distance = 'Distance: ' + variable_list["distance"]["text"]
+        duration = 'Duration: ' + variable_list["duration"]["text"]
+        output = begin + '\n' + end + '\n' + distance
+        # if user doesn't know that default mode is driving
+        if 'mode' not in params:
+            mode = 'Mode of Transport: Driving'
+            output += '\n' + mode
+
+        # determines if fare information is available
+        try:
+            fare = ('Fare: ' + variable_list["fare"]["currency"]
+                    + variable_list["fare"]["text"])
+            output += '\n' + fare
+        except (KeyError, IndexError):
+            pass
+
+        # determines if traffic duration information is available
+        try:
+            traffic_duration = ('Duration in traffic: '
+                                + variable_list["duration_in_traffic"]
+                                ["text"])
+            output += '\n' + traffic_duration
+        except (KeyError, IndexError):
+            output += '\n' + duration
+
+        # bot sends commute information to user
+        self.send_info(message, output, client)
+
+    # creates parameters for HTTP request
+    def parse_pair(self, content_list):
+        result = {}
+        for item in content_list:
+            # enables key value pair
+            org = item.split('=')
+            # ensures that invalid inputs are not entered into url request
+            if len(org) != 2:
+                continue
+            key, value = org
+            result[key] = value
+        return result
+
+    def receive_response(self, params, message, client):
+        def validate_requests(request):
+            if request.status_code == 200:
+                return request.json()
+            else:
+                self.send_info(message,
+                               "Something went wrong. Please try again."
+                               + " Error: {error_num}.\n{error_text}"
+                               .format(error_num=request.status_code,
+                                       error_text=request.text), client)
+                return
+        r = requests.get('https://maps.googleapis.com/maps/api/'
+                         + 'distancematrix/json', params=params)
+        result = validate_requests(r)
+        return result
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+        content_list = original_content.split()
+
+        if "help" in content_list:
+            self.send_info(message, self.help_info, client)
+            return
+
+        params = self.parse_pair(content_list)
+        params['key'] = self.api_key
+        self.add_time_to_params(params)
+
+        rjson = self.receive_response(params, message, client)
+        if not rjson:
+            return
+
+        self.get_send_content(rjson, params, message, client)
+
+handler_class = CommuteHandler
+handler = CommuteHandler()
```

Comment:
If we only need this handler for the tests, we not create this at module scope.  We can address this in a follow up PR, though, as needed, once we figure out a more long term plan for handling bot tests.